server/server.go

```diff
@@ -765,7 +765,7 @@ func (s *Server) shouldTrackSubscriptions() bool {
 // Lock should be held on entry.
 func (s *Server) registerAccount(acc *Account) {
 	if acc.sl == nil {
-		acc.sl = NewSublist()
+		acc.sl = NewSublist(s.opts != nil && !s.opts.NoSublistCache)
```

Comment:
Not sure you intended to create a sublist with no cache if s.opts is nil. Also, to avoid data races, I would replace this with something like.

Note: I ran all tests and checked if s.opts could be nil and it is never...
```
opts := s.getOpts()
useCache := opts == nil || !opts.NoSublistCache
acc.sl = NewSublist(useCache)
```
or even, to be more explicit
```
opts := s.getOpts()
if opts != nil && opts.NoSublistCache {
   acc.sl = NewSublistNoCache()
} else {
   acc.sl = NewSublistWithCache()
}
```

Reply:
Yes I did, one test failed under server for split buffers.

Reply:
I can do that. Will make the change.

Reply:
Right.. this is because I did the test on master where s.gacc would still have the sublist already created.. So I would add a comment that says why we check for opts == nil.